providers/shibboleth/shibboleth_client.go

```diff
@@ -23,52 +18,33 @@ import (
 //SPClient implements a client for shibboleth and the saml library
 type SPClient struct {
 	config *model.ShibbolethConfig
-	samlSP *samlsp.Middleware
 }
 
-func (sp *SPClient) initializeSPClient(configToSet *model.ShibbolethConfig) error {
+func (spclient *SPClient) initializeSPClient(configToSet *model.ShibbolethConfig) error {
 
-	var idpURL string
 	var privKey *rsa.PrivateKey
 	var cert *x509.Certificate
 	var err error
 	var ok bool
 
-	sp.config = configToSet
+	spclient.config = configToSet
 
-	if configToSet.IDPMetadataURL == "" {
```

Comment:
All this theoretically dead code you removed to simplify configuration - can you NOT remove it?
There is a chance that in v1.6, someone is actually using IDPMetadataFilePath or SPSelfSignedCertFilePath or SPSelfSignedKeyFilePath

Reply:
@cjellick I remember checking that file paths weren't working, cattle code was missing something. I'll check what it was